planner/core/preprocess.go

```diff
@@ -356,6 +389,39 @@ func (p *preprocessor) checkBindGrammar(originNode, hintedNode ast.StmtNode, def
 			return
 		}
 	}
+
+	// Check the bind operation is not on any temporary table.
+	var resNode ast.ResultSetNode
+	switch n := originNode.(type) {
+	case *ast.SelectStmt:
+		resNode = n.From.TableRefs
+	case *ast.DeleteStmt:
+		resNode = n.TableRefs.TableRefs
+	case *ast.UpdateStmt:
+		resNode = n.TableRefs.TableRefs
+	case *ast.InsertStmt:
+		resNode = n.Table.TableRefs
+	}
+	if resNode != nil {
+		tblNames := extractTableList(resNode, nil, false)
+		for _, tn := range tblNames {
+			tbl, err := p.tableByName(tn)
+			if err != nil {
+				// If the operation is order is: drop table -> drop binding
+				// The table doesn't  exist, it is not an error.
+				if terror.ErrorEqual(err, infoschema.ErrTableNotExists) {
```

Comment:
Is not this check silent all `drop binding` statements, including ones that trying to drop bindings for non-exsist tables? BTW, should add drop binding cases.

Reply:
This will not silent all `drop binding` statements error, it just silent the `infoschema.ErrTableNotExists` error.

The behaviour is the same, prior to this change, we do not check table existence, and now we check the table but ignore the `ErrTableNotExists` error

Check temporary table for `drop binding` is not necessary, because we forbid the `create binding`, then there are no temporary tables in `drop binding` ...